### PR TITLE
[TwigBridge] Support for Twig 3

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

I occasionally test my projects against dev branches of the libraries I use. It's probably a bit early for this PR, but since I had to patch Symfony anyway in order to test Twig 3 in my projects, I though I might as well share the branch with you.

BC break: I had to add return types to several non-final classes of the Twig bridge. This will break applications that override these classes without adding the return types.

This PR is marked as a draft because Twig 3 is still a moving target.

### CODE REVIEW

**Comment by fabpot** [comment]
See #33269

**Comment by fabpot** [comment]
Now that @33269 has been merged, I think we can finish this one @derrabus. Twig 3.0 will be released *before* Symfony 5, so this is safe and will allow us to spot issues sooner than later. Thank you.

File: src/Symfony/Bridge/Twig/Form/TwigRendererEngine.php [hunk 1]
```diff
@@ -149,7 +149,7 @@ protected function loadResourcesFromTheme(string $cacheKey, &$theme)
     {
         if (!$theme instanceof Template) {
             /* @var Template $theme */
-            $theme = $this->environment->loadTemplate($theme);
+            $theme = $this->environment->load($theme)->unwrap();
```
Comment (derrabus) [inline src/Symfony/Bridge/Twig/Form/TwigRendererEngine.php]: Can/should we backport this to 4.4?

Comment (fabpot) [reply src/Symfony/Bridge/Twig/Form/TwigRendererEngine.php]: Yes

**Comment by derrabus** [comment]
Ready.

Also, I'm wondering if we should backport the composer.json changes (except for the bridge, of course) to 4.4. An Symfony 4.4 application with TwigBridge 5.0 should technically be able to run Twig 3.

**Comment by fabpot** [comment]
Symfony 4.4 still supports 1.x, so it cannot support 1.x/2.x and 3.0 at the same time.

File: src/Symfony/Bridge/Twig/Form/TwigRendererEngine.php (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Bridge/Twig/Form/TwigRendererEngine.php]: #33273

**Comment by fabpot** [comment]
I've just merged 4.4 into master so that this one can be rebased.

**Comment by nicolas-grekas** [comment]
We should also replace all `@final since Symfony 4.4` by `@final` here.

**Comment by derrabus** [comment]
Rebased.

**Comment by derrabus** [comment]
> We should also replace all `@final since Symfony 4.4` by `@final` here.

Done.

File: src/Symfony/Bridge/Twig/Extension/YamlExtension.php [hunk 2]
```diff
@@ -30,7 +30,7 @@ class YamlExtension extends AbstractExtension
      *
      * @return TwigFilter[]
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Extension/YamlExtension.php]: we can now remove all such lines, `@inheritdoc` is enough

File: src/Symfony/Bridge/Twig/Extension/AssetExtension.php [hunk 3]
```diff
@@ -20,7 +20,7 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  *
- * @final since Symfony 4.4
+ * @final
```
Comment (fabpot) [inline src/Symfony/Bridge/Twig/Extension/AssetExtension.php]: Can't we move it to a real final class? `final class AssetExtension ...`

Comment (derrabus) [reply src/Symfony/Bridge/Twig/Extension/AssetExtension.php]: Certainly. @nicolas-grekas?

Comment (stof) [reply src/Symfony/Bridge/Twig/Extension/AssetExtension.php]: well, we should do it in master indeed.

**Comment by derrabus** [comment]
Should I also remove the `getName()` implementations from the Twig extensions? IIRC, we only need them for Twig 1 which has been dropped on master. Do we use them for anything else?

**Comment by derrabus** [comment]
Also, I will add return type declarations to all `@final`classes where possible.

**Comment by fabpot** [comment]
`getName()` can be removed

**Comment by derrabus** [comment]
Ready.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @derrabus.

**Pull request merged**